base/ca/shared/webapps/ca/agent/ca/ProfileReview.template

```diff
@@ -43,6 +43,16 @@ function addEscapes(str)
     return outStr;
 }
 
+function validateForm(form) {
+  pathLength = form.basicConstraintsPathLen;
+
+  if(pathLength != null && !Number.isInteger(parseInt(pathLength.value))) {
```

Comment:
`Number.isInteger` wrapping `parseInt` is excessive and redundant. :)

Reply:
> This validation should occur on the server side. Validation on the client side is fine for a message to the user, but ultimately it should be occurring on the server side.
> 

There is server side validation already included. The web UI throws this message when you try to enter anything funky
````
 The Certificate System has encountered an unrecoverable error.

Error Message:
java.lang.NumberFormatException: For input string: ""

Please contact your local administrator for assistance
````
The CVE is to prevent execution of malicious JS on the client browser. So, my patch is trying to clean out before sending the values to the server.

Also, note that I couldn't test this with anyother value other than `-1`. I filed a bug: https://pagure.io/dogtagpki/issue/3179

> **Edit**: Additionally, you can use the [number typed input element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number) instead of this client-side validation. However, you still need server-side validation.

I can try that

Reply:
Then it isn't clear to me where _exactly_ the vulnerability occurs? Is it because the error message is displayed incorrectly (getting interpreted as HTML)? If so, was it fixed by #434? If not, we should fix the site that displays the error message.

Otherwise, I really don't see how this is worthy of a CVE: user types at computer and attacks themselves. Closed->NOTABUG. You don't need RHCS for that. You can do that with bash or literally anything else. Or directly on the web console. 

Is this XSS stored? Does it get saved for an agent to stumble upon later? Then you need to sanitize the incorrect value before saving it in the database and also sanitize where it gets displayed to the agent. But this isn't the proper fix for it.

Reply:
Yeah, good point. Someone could be using the pki cli command for some of this stuff.

Reply:
Also perhaps he is using the isInteger to handle the case if parseInt returns the Nan value, which I guess is not a number.

Reply:
Ah I figured this out. The server endpoint doesn't do any CSRF protection and also responds to GET requests. This truly makes it a reflected XSS, which is bad. The fix is three part:

 - Sanitize the output error message,
 - Enable CSRF protection,
 - Optionally handle only POST requests.